### ISSUE TITLE
Inconsistent timestamps across updates in SegmentTimeline

### ISSUE THREAD

**Issue by joeyparrish** [Dash-Industry-Forum/dash-live-source-simulator#91]
When playing this live stream:
https://livesim.dashif.org/livesim/segtimeline_1/utc_head/testpic_6s/Manifest.mpd

I find that the timestamps in the SegmentTimeline are inconsistent across updates.  For example:

```xml
<SegmentTimeline>
<S d="288768" t="75776683106304" />
<S d="287744" />
```

The first segment starts at 75776683106304 and ends at 75776683106304 + 288768 = 75776683395072.  After updating the manifest, the timestamp jumps:

```xml
<SegmentTimeline>
<S d="287744" t="75776683392000" />
```

75776683395072 jumped backward to 75776683392000, a difference of 3072 timescale units.  The timescale is 48000, so the discrepancy is 3072 / 48000 seconds, or 64 ms.

It seems like this sort of thing shouldn't happen, especially given that the segment template is based on `$Time$`.  When the first timestamp changes, every other segment timestamp changes, too.  With every segment start time and URL changing at once, and with no startNumber to base things on, the segments before and after the update have nothing in common with each other.

Here are two complete manifests captured from the live sim from which I took the snippets above:

 - [5.mpd.txt](https://github.com/Dash-Industry-Forum/dash-live-source-simulator/files/4047463/5.mpd.txt)
 - [6.mpd.txt](https://github.com/Dash-Industry-Forum/dash-live-source-simulator/files/4047462/6.mpd.txt)

Is this a bug in the live sim?  If not, how should I interpret the change in timestamps if there are no matching segment timestamps from one manifest update to the next?

P.S. Thanks for creating and maintaining this very useful tool!

**Comment by TobbeEdgeware** [comment]
Hi Joey, 

This seems to be bug indeed. The time error is three AAC frames (1024 samples in each).

As you say, for SegmentTimeline with $Time$ without t-value, there is no other way to calculate the right URL than taking the t-value of a previous \<S\>-element and add the duration. For SegmentTimeline with $Number$ (the favourite of Shaka packager?), the URL-situation is of course different, although the timestamps shall be accurate and not change in that case as well.

Thanks for the encouraging words. I'll look a bit deeper into this once I get a few spare hours.

**Comment by joeyparrish** [comment]
Thanks!

**Comment by TobbeEdgeware** [comment]
Sorry for the delay, but I've finally found the issue. It was a helper file with timestamps for the segments which was corrupt for this instance of the live simulator. I've updated the that file to a consistent one and then the buggy behaviour was no longer there (a nice thing about simulating live is that I could reconstruct the exact same MPD that you reported by choosing the same wall-clock time).
